src/client.rs

```diff
@@ -486,6 +512,55 @@ impl Client {
         self._pull_manifest(image).await
     }
 
+    /// Pull an image manifest from the remote OCI Distribution service.
+    ///
+    /// If the connection has already gone through authentication, this will
+    /// use the bearer token. Otherwise, this will attempt an anonymous pull.
```

Comment:
Why do we need to pull this out into a separate function? Why not wrap `auth` as an `Option<&RegistryAuth>` instead?

Reply:
I was using the same pattern as _pull_manifest. I assume that pattern is there so that the external API mandates a RegistryAuth, but internally that may not be required (as it's already auth has already been done).

Reply:
Is this conversation still relevant? I think we can mark that as resolved

Reply:
marking as resolved based on my previous comment